tests/Aspire.Workload.Tests/StarterTemplateProjectNamesTests.cs

```diff
@@ -1,50 +1,78 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Aspire.Components.Common.Tests;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Aspire.Workload.Tests;
 
-public class StarterTemplateProjectNamesTests : WorkloadTestsBase
+public abstract class StarterTemplateProjectNamesTests : WorkloadTestsBase
 {
-    public StarterTemplateProjectNamesTests(ITestOutputHelper testOutput)
+    private readonly string _testType;
+    public StarterTemplateProjectNamesTests(string testType, ITestOutputHelper testOutput)
         : base(testOutput)
     {
+        _testType = testType;
     }
 
-    public static TheoryData<string, string> ProjectNamesWithTestType_TestData()
+    public static IEnumerable<object[]> ProjectNamesWithTestType_TestData()
     {
-        var data = new TheoryData<string, string>();
-        foreach (var testType in TestFrameworkTypes)
+        foreach (var name in GetProjectNamesForTest())
         {
-            foreach (var name in GetProjectNamesForTest())
-            {
-                data.Add(name, testType);
-            }
+            yield return [name];
         }
-        return data;
     }
 
     [Theory]
     [MemberData(nameof(ProjectNamesWithTestType_TestData))]
-    public async Task StarterTemplateWithTest_ProjectNames(string prefix, string testType)
+    [RequiresSSLCertificate("Needs dashboard, web front end access")]
+    public async Task StarterTemplateWithTest_ProjectNames(string prefix)
     {
-        string id = $"{prefix}-{testType}";
+        string id = $"{prefix}-{_testType}";
         string config = "Debug";
 
         await using var project = await AspireProject.CreateNewTemplateProjectAsync(
             id,
             "aspire-starter",
             _testOutput,
             BuildEnvironment.ForDefaultFramework,
-            $"-t {testType}");
+            $"-t {_testType}");
 
         await using var context = PlaywrightProvider.HasPlaywrightSupport ? await CreateNewBrowserContextAsync() : null;
         _testOutput.WriteLine($"Checking the starter template project");
         await AssertStarterTemplateRunAsync(context, project, config, _testOutput);
 
         _testOutput.WriteLine($"Checking the starter template project tests");
-        await AssertTestProjectRunAsync(project.TestsProjectDirectory, testType, _testOutput, config);
+        await AssertTestProjectRunAsync(project.TestsProjectDirectory, _testType, _testOutput, config);
+    }
+}
+
+// Individual class for each test framework so the tests can run in separate helix jobs
```

Comment:
Why are these split, but BuildAndRunTemplateTests aren't?

Reply:
These had a natural split based on the template argument for the test type. I can split the others in a follow up based on how long they take to run. But that timing will be affected once we start using the `actions/cache`.